src/Native/Runtime/amd64/InteropThunksHelpers.S

```diff
@@ -4,19 +4,38 @@
 
 .intel_syntax noprefix
 #include <unixasmmacros.inc>
-
-//TODO: Implement these stub method for non-windows
+#include <AsmOffsets.inc>         // generated by the build from AsmOffsets.cpp
+#define POINTER_SIZE 8
 
 LEAF_ENTRY RhpCommonStub, _TEXT
-    int 3
+    push_nonvol_reg rdi
+    mov    r11, [r10]
+#if defined(__APPLE__)
+    mov    rdi, qword ptr [rip + _tls_thunkData@TLVP]
+    call   qword ptr [rdi]
```

Comment:
Did you find a documentation somewhere that says that this preserves all argument registers?